main/settings.hpp

```diff
@@ -6,8 +6,8 @@
 namespace settings
 {
 // HTTP stuff
-constexpr auto EndpointName = "radon";
-constexpr auto EndpointPort = 9003;
+constexpr auto EndpointName = "https://pwr-meter-pulse-gw-19i.svc.n39.eu";
+constexpr auto EndpointPort = 443;
```

Comment:
Vorher stand dort nur radon, jetzt steht dort `https://` - ist das korrekt?

Reply:
Das werde ich heute Abend testen.

Oder sind `http://` bzw `https://` nur syntaktischer Zucker für den eigentlichen Port dahinter?

Reply:
in der URI gibt `https` das Schema an und ist damit notwendig, um den Rest zu interpretieren.

Der Rest hängt von der Implementierung ab: Wenn Dein Client URIs auswertet, wird der daraus ableiten, dass Port 443 und Verschlüsselung nach HTTPS zu verwenden sind.

Die URI `https://pwr-meter-pulse-gw-19i.svc.n39.eu/pulse` enthält folgende Informationen:
* Protokoll HTTPS
* Host pwr-meter-pulse-gw-19i.svc.n39.eu/
* (implizit) Port 443
* Pfad `/pulse`

Alles andere hängt von Deinem Client ab. Aber da Du auch den Pfad einzeln angibst, sieht es so aus, als wollte der Client hier einen Hostname haben, der sich mit DNS auflösen lässt?

Siehe auch:
* https://de.wikipedia.org/wiki/Uniform_Resource_Identifier
* https://de.wikipedia.org/wiki/Hypertext_Transfer_Protocol
* https://de.wikipedia.org/wiki/Hypertext_Transfer_Protocol_Secure

Reply:
Wenn ich in der [Doc](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/esp_http_client.html#https-request) von espressif schaue, dann nutzt er das Scheme auch, wenn die URL mit `https` anfängt.

Allerdings wird dieser [struct](https://github.com/espressif/esp-idf/blob/d6d8324ad9b957f85ecb323ec0db12c4d03a93c4/components/esp_http_client/include/esp_http_client.h#L101) gefüllt. Dort nutze ich die members `host`, `port` und `path`.

Ich werde es heute Abend probieren

Reply:
Nach der Doc scheint es am einfachsten, die komplette URL in der Konfiguration anzugeben. Ansonsten musst Du Verschlüsselung explizit aktivieren.